UnityProject/Assets/Scripts/Systems/GhostRoles/GhostRoleManager.cs

```diff
@@ -223,7 +223,12 @@ private GhostRoleResponseCode VerifyPlayerCanQueue(ConnectedPlayer player, uint
 
 			GhostRoleServer role = serverAvailableRoles[key];
 
-			if (PlayerList.Instance.CheckJobBanState(player.UserId, role.RoleData.TargetOccupation.JobType) == false)
+			if (role.RoleData.TargetOccupation != null && PlayerList.Instance.CheckJobBanState(player.UserId, role.RoleData.TargetOccupation.JobType) == false)
+			{
+				return GhostRoleResponseCode.JobBanned;
+			}
+
+			if (role.RoleData.TargetAntagonist != null && PlayerList.Instance.CheckJobBanState(player.UserId, role.RoleData.TargetAntagonist.AntagJobType) == false)
```

Comment:
if only we reparented Antagonist and Occupation to be children of the same class :eyes: